plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -317,6 +319,13 @@ public PluginCompatReport testPlugins()
                         }
                         report.save(config.reportFile);
                     }
+
+                    if (status != TestStatus.SUCCESS) {
+                        failed = true;
+                        if (config.isFailOnError()) {
```

Comment:
Why not launching the exception here?

Reply:
The output will not be written to the report then. It happens only in the end of the method